src/components/ProjectSettingsTabs/GeneralTab/Fields/RepoConfigField.tsx

```diff
@@ -0,0 +1,251 @@
+} from "gql/generated/types";
+import {
+  ATTACH_PROJECT_TO_REPO,
+  DETACH_PROJECT_FROM_REPO,
+} from "gql/mutations";
+
+interface ModalProps {
+  onCancel: () => void;
+  onConfirm: (formUpdate: any) => void;
+  open: boolean;
+}
+
+export const MoveRepoModal: React.FC<ModalProps> = ({
+  onCancel,
+  onConfirm,
+  open,
+}) => {
+  const [formState, setFormState] = useState({});
+  const [hasError, setHasError] = useState(false);
+  const isDisabled =
+    Object.keys(formState).length === 0 ||
+    Object.values(formState).some((input) => input === "") ||
+    hasError;
+
+  return (
+    <ConfirmationModal
+      buttonText="Move Repo"
+      data-cy="move-repo-modal"
+      onCancel={onCancel}
+      onConfirm={() => onConfirm(formState)}
+      open={open}
+      submitDisabled={isDisabled}
+      title="Move Repo"
+      variant="danger"
+    >
+      Select an existing repository or add a new one.
+      {/* TODO: Add select component upon completion of EVG-15037 */}
+      <SpruceForm
+        formData={formState}
+        onChange={({ formData, errors }) => {
+          setHasError(errors.length > 0);
+          setFormState(formData);
+        }}
+        schema={modalFormDefinition.schema}
+        uiSchema={modalFormDefinition.uiSchema}
+      />
+    </ConfirmationModal>
+  );
+};
+
+export const AttachDetachModal: React.FC<{
+  handleClose: () => void;
+  open: boolean;
+  projectId: string;
+  repoName: string;
+  repoOwner: string;
+  shouldAttach: boolean;
+}> = ({ handleClose, open, projectId, repoName, repoOwner, shouldAttach }) => {
+  const dispatchToast = useToastContext();
+
+  const [attachProjectToRepo] = useMutation<
+    AttachProjectToRepoMutation,
+    AttachProjectToRepoMutationVariables
+  >(ATTACH_PROJECT_TO_REPO, {
+    variables: { projectId },
+    onCompleted() {
+      dispatchToast.success("Successfully attached to repo");
+    },
+    onError(err) {
+      dispatchToast.error(
+        `There was an error attaching the project: ${err.message}`
+      );
+    },
+    refetchQueries: ["ProjectSettings", "RepoSettings"],
+  });
+
+  const [detachProjectFromRepo] = useMutation<
+    DetachProjectFromRepoMutation,
+    DetachProjectFromRepoMutationVariables
+  >(DETACH_PROJECT_FROM_REPO, {
+    variables: { projectId },
+    onCompleted() {
+      dispatchToast.success("Successfully detached from repo");
+    },
+    onError(err) {
+      dispatchToast.error(
+        `There was an error detaching the project: ${err.message}`
+      );
+    },
+    refetchQueries: ["ProjectSettings", "RepoSettings"],
+  });
+
+  return (
+    <ConfirmationModal
+      buttonText={shouldAttach ? "Attach" : "Detach"}
+      data-cy="attach-repo-modal"
+      onCancel={handleClose}
+      onConfirm={() => {
+        if (shouldAttach) {
+          attachProjectToRepo();
+        } else {
+          detachProjectFromRepo();
+        }
+        handleClose();
+      }}
+      open={open}
+      title={`Are you sure you want to ${
```

Comment:
![image](https://user-images.githubusercontent.com/26798134/154510386-798a0a95-f335-4000-b17d-31f1ba0ae610.png)
When I clicked "Save Changes on Page" for the General Section it now detaches me from repo, regardless of whether or not I had clicked into the modal.

Reply:
That's funny, this is actually a consequence of EVG-16081, which began as "SaveProjectSettingsForSection incorrectly sets useRepoSettings." The behavior we're seeing there involves `useRepoSettings` being returned as `false` whenever the save resolver is called. You can verify that we're not _actually_ detaching from the repo by the fact that there's no network request being made to the `DetachProjectFromRepo` resolver.